packages/uswds-core/src/styles/mixins/helpers/set-icon-from-bg.scss

```diff
@@ -0,0 +1,14 @@
+@use "../../settings" as *;
+@use "../../functions/color/is-color-dark" as *;
+@use "../../mixins/general/add-background-svg" as *;
+
+/// Set an icon depending on the background color.
+/// @param {type} $background-color - The elements background color.
+/// @param {type} $icon - The icon that will be set. Requires a variant in `usa-icons-bg` directory.
+/// @output - A light or dark icon depending on the $background-color.
+@mixin set-icon-from-bg($background-color, $icon) {
+  $is-darkmode: is-color-dark($background-color);
+  $icon: if($is-darkmode, "usa-icons-bg/#{$icon}--white", "usa-icons/#{$icon}");
+
```

Comment:
Wondering if it would be beneficial to add an error message for when the `--white` variant of the icon doesn't exist. Alternatively, maybe we could provide a fallback with a warning?

Reply:
Makes sense to me! It is internal and limited use, so it makes sense for it to be added a follow-up.